src/test/java/it/gov/pagopa/arc/dto/mapper/gpd/GPDPaymentNoticePayloadDTO2PaymentNoticeDetailsDTOMapperTest.java

```diff
@@ -0,0 +1,152 @@
+        //when
+        PaymentNoticeDetailsDTO result = gpdPaymentNoticePayloadDTO2PaymentNoticeDetailsDTOMapper.map(null);
+        //then
+
+        Assertions.assertNull(result);
+    }
+
+    @Test
+    void givenNullGPDPaymentOptionPayloadDTOWhenMapThenReturnNull() {
+        //given
+        GPDPaymentNoticePayloadDTO gpdPaymentNoticePayloadDTO = GPDPaymentNoticePayloadDTOFaker.mockInstance("DUMMY_ORGANIZATION_FISCAL_CODE");
+        gpdPaymentNoticePayloadDTO.setPaymentOption(null);
+        //when
+        PaymentNoticeDetailsDTO result = gpdPaymentNoticePayloadDTO2PaymentNoticeDetailsDTOMapper.map(gpdPaymentNoticePayloadDTO);
+        //then
+
+        Assertions.assertNull(result);
+    }
+
+    @Test
+    void givenEmptyGPDPaymentOptionPayloadDTOWhenMapThenReturnNull() {
+        //given
+        GPDPaymentNoticePayloadDTO gpdPaymentNoticePayloadDTO = GPDPaymentNoticePayloadDTOFaker.mockInstance("DUMMY_ORGANIZATION_FISCAL_CODE");
+        gpdPaymentNoticePayloadDTO.setPaymentOption(new ArrayList<>());
+        //when
+        PaymentNoticeDetailsDTO result = gpdPaymentNoticePayloadDTO2PaymentNoticeDetailsDTOMapper.map(gpdPaymentNoticePayloadDTO);
+        //then
+
+        Assertions.assertNull(result);
+    }
+
+    @Test
+    void givenNullGPDTransferPayloadDTOWhenMapThenReturnNull() {
+        //given
+        GPDPaymentNoticePayloadDTO gpdPaymentNoticePayloadDTO = GPDPaymentNoticePayloadDTOFaker.mockInstance("DUMMY_ORGANIZATION_FISCAL_CODE");
+        gpdPaymentNoticePayloadDTO.getPaymentOption().get(0).setTransfer(null);
+        //when
+        PaymentNoticeDetailsDTO result = gpdPaymentNoticePayloadDTO2PaymentNoticeDetailsDTOMapper.map(gpdPaymentNoticePayloadDTO);
+        //then
+
+        Assertions.assertNull(result);
+    }
+
+    @Test
+    void givenEmptyGPDTransferPayloadDTOWhenMapThenReturnNull() {
+        //given
+        GPDPaymentNoticePayloadDTO gpdPaymentNoticePayloadDTO = GPDPaymentNoticePayloadDTOFaker.mockInstance("DUMMY_ORGANIZATION_FISCAL_CODE");
+        gpdPaymentNoticePayloadDTO.getPaymentOption().get(0).setTransfer(new ArrayList<>());
+        //when
+        PaymentNoticeDetailsDTO result = gpdPaymentNoticePayloadDTO2PaymentNoticeDetailsDTOMapper.map(gpdPaymentNoticePayloadDTO);
+        //then
+
+        Assertions.assertNull(result);
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideEnumMappings")
+    void givenGPDPaymentNoticeStatusWhenMapThenReturnPaymentNoticeDetailsStatus(
+            GPDPaymentNoticeStatus gpdPaymentNoticeStatus,
+            PaymentNoticeDetailsStatus expectedPaymentNoticeDetailsStatus) {
+
+        PaymentNoticeDetailsStatus result =
+                gpdPaymentNoticePayloadDTO2PaymentNoticeDetailsDTOMapper
+                        .gPDPaymentNoticeStatusToPaymentNoticeDetailsStatus(gpdPaymentNoticeStatus);
+
+        Assertions.assertEquals(expectedPaymentNoticeDetailsStatus, result);
+    }
+
+    private static Stream<Arguments> provideEnumMappings() {
```

Comment:
instead of this, if we plan to have always the same name, we could just call the method for each GPDPaymentNoticeStatus value, verifying that the name is equal

for(GPDPaymentNoticeStatus e in GPDPaymentNoticeStatus.values){
 PaymentNoticeDetailsStatus result = map...
 Assertequals(e.name(), result.name())

This way you will test for each GPD values (also eventually new enum)